DDCore/src/XML/DocumentHandler.cpp

```diff
@@ -328,13 +328,20 @@ Document DocumentHandler::load(Handle_t base, const XMLCh* fname, UriReader* rea
 
 /// Load XML file and parse it using URI resolver to read data.
 Document DocumentHandler::load(const std::string& fname, UriReader* reader) const   {
+  auto _clean_fname = [](const std::string& s) -> std::string {
+    const std::string& temp = getEnviron(s);
+    std::string temp2 = undressed_file_name(temp.empty() ? s : temp);
+    if ( strncmp(temp2.c_str(),"file:",5) == 0 ) return temp2.substr(5);
+    return temp2;
+  };
+  auto fname_clean = _clean_fname(fname);
```

Comment:
```suggestion
  const auto fname_clean = [](const std::string& s) -> std::string {
    const std::string& temp = getEnviron(s);
    std::string temp2 = undressed_file_name(temp.empty() ? s : temp);
    if ( strncmp(temp2.c_str(),"file:",5) == 0 ) return temp2.substr(5);
    return temp2;
  }();
```
We don't need to give the lambda a name in this case, we can also invoke it immediately. The whole thing could probably do with a brief comment describing in english what it does / is supposed to do.

Reply:
just use the _clean_fname function that already exists in this file?